common/metrics/tally_client.go

```diff
@@ -113,8 +121,10 @@ func (m *TallyClient) RecordTimer(scopeIdx int, timerIdx int, d time.Duration) {
 // metric name
 func (m *TallyClient) RecordDistribution(scopeIdx int, timerIdx int, d int) {
 	dist := time.Duration(d * distributionToTimerRatio)
-	name := string(m.metricDefs[timerIdx].metricName)
-	m.childScopes[scopeIdx].Timer(name).Record(dist)
+	def := m.metricDefs[timerIdx]
+	name := string(def.metricName)
+	buckets, _ := m.perUnitBuckets[def.unit]
```

Comment:
buckets := m.per..., since you don't check the exists or not result.
Also, where do we ensure that the def.unit is present in the map? Or is it possible to be nil and we won't care if it is nil?

Reply:
If it is nil, default buckets set will be used.